content/articles/constructors_in_kotlin/index3.md

```diff
@@ -0,0 +1,205 @@
+
+### Table of Contents
+- [What is a constructor?](#what-is-a-constructor)
+- [How to use constructors to create an object](#how-to-use-constructors-to-create-an-object)
+- [Types of constructors in Kotlin](#types-of-constructors-in-kotlin)
+- [Init blocks](#init-blocks)
+- [Best practices for object-oriented programming](#best-practices-for-object-oriented-programming)
+
+### Prerequisites
+To fully understand the article, you are required to have the following:
+
+- The Java Development Kit (JDK) installed on your computer
+- A basic understanding of programming languages
+- The IntelliJ code editor
+
+### What is a constructor?
+A constructor is a block of code that is similar to a procedure or a method, usually declared alongside the class. They are mostly used when you need to initialize variables while creating an object.
+
+### Types of constructors in Kotlin
+There are two different kinds of constructors in Kotlin, which are:
+
+- The primary constructor
+- The secondary constructor
+
+#### How to use constructors to create an object
+The primary constructor must be defined with the same name as the class, followed by parentheses. However, the secondary constructor must be expressed with the `constructor` keyword. Note: If the programmer does not define the constructor, the compiler is designed to generate a default one.
+
+#### Primary constructors
+The primary constructor is a section of the class header that comes after the class name, using the `constructor` keyword. The example below exemplifies the definition:
+
+```kotlin
+class student constructor(name: String, age:Int, Email:String){ 
+     //code to be processed
+}
+```
+The constructor keyword can be ignored if the primary constructor has no annotations or accessibility modifiers:
+
+The example below explains:
+```kotlin
+class AtmMachine(_moneyInMachine:Boolean,_location:String){
+    //code to be exercuted
+}
+```
+In the preceding example, we don't use the `constructor` keyword. By default, all constructors are public which means they are visible everywhere in the class. Therefore we don't have to include the `constructor` keyword, as shown below:
+
+```kotlin
+class student(name:string,age:Int,Email:string){ }
+```
+
+#### Secondary constructor
+In Kotlin, a secondary constructor is created using the `constructor` keyword. In addition, the secondary constructor can be used to call the primary constructor. The example below shows how you can declare two constructors with a student having two parameters (name and year). 
+
+```kotlin
+class student{ 
+    constructor(name: Int){ 
+
+    } 
+    constructor(year:Int, id: Int){ 
+
+    } 
+}
+```
+
+Also, the secondary constructor can assign values to fields while the class is instantiated. The example below illustrates this property:
+
+```kotlin
+class myStudent{ 
+    constructor(name:String,id:Int){
+         println("Name = ${name}") 
+         println("Id = ${id}")
```

Comment:
You aren't assigning any values to fields in this example unlike what you said at like 67. You are printing values here.

Reply:
@john-amiscaray the values alex and 200 are assigned

Reply:
To me, it seems that in this code snippet you are only printing the following to the console:

```kotlin
Name = alex
Id = 200
```

In your explanation for this code, you said you are assigning values to fields in this object but it looks to me that you are only printing them. Let me know if there is a misunderstanding about this on my part since I've only picked up Kotlin fairly recently.